File: src/compiler/interpreter/AssignmentTypeException.java
package compiler.interpreter;

import compiler.lexer.token.Token;
import compiler.lexer.token.TypeToken;
import compiler.parser.UserException;

public class AssignmentTypeException extends UserException {
  public AssignmentTypeException(Token lval, Token rval, TypeToken lvalType, TypeToken rvalType) {
    super(lval.getValue() + " is of type " + lvalType.getValue() + " and cannot be assigned to a type of " + rvalType.getValue(), rval);
  }
}

File: src/compiler/interpreter/AssignmentTypeException.kt
package compiler.interpreter

import compiler.parser.Language.Token
import compiler.parser.Language.Token.KeywordToken.TypeToken
import compiler.parser.UserException

class AssignmentTypeException(lval: Token, rval: Token, lvalType: TypeToken, rvalType: TypeToken) :
        UserException("${lval.str} is of type ${lvalType.str} and cannot be assigned to a type of ${rvalType.str}", rval)
